ui/components/multichain/account-list-menu/account-list-menu.js

```diff
@@ -73,7 +73,7 @@ const ACTION_MODES = {
  * @param internalAccounts - internal accounts
  * @returns merged accounts list with balances and internal account data
  */
-const mergeAccounts = (accountsWithBalances, internalAccounts) => {
+export const mergeAccounts = (accountsWithBalances, internalAccounts) => {
   return accountsWithBalances.map((account) => {
     const internalAccount = internalAccounts.find(
```

Comment:
I realize this is not your code so feel free to ignore but I couldn't help but notice some glaring inefficiencies with this code....

The use of `.find` that is called from on every iterations is not needed and is resulting on O(n*2) code. If we created a lookup map from address to account we could swap this for a `.get` call which is constant time. The code would look something like this...

```ts
export const mergeAccounts = (accountsWithBalances, internalAccounts) => {
  const internalAccountsMap = new Map(internalAccounts.map(acc => [acc.address, acc]));

  return accountsWithBalances.map((account) => {
    const internalAccount = internalAccountsMap.get(account.address);
    if (internalAccount) {
      return {
        ...account,
        ...internalAccount,
        name: internalAccount.metadata?.name || account.name,
        keyring: internalAccount.metadata?.keyring,
        label: getAccountLabel(
          internalAccount.metadata?.keyring?.type,
          internalAccount,
        ),
      };
    }
    return account;
  });
};

```

Reply:
Out of scope, but I agree we should improve this!

Reply:
when we move to create the new selector we can implement this change.

Reply:
I agree, we should follow this PR with a small PR to move this logic into a selector. It will be easier to update the code using the selector once Identities are changed.